HHS/Head-Start-TTADP
# [prod] Automatically delete creator & manager notes on approval, make sure reports in "needs action" have editable creator notes

## Description of change
- Delete all historical creator & manager notes
- Automatically delete notes from reports on approval
- The creator notes field is now editable when the report is "needs action"

## How to test

Create a report, submit. Make sure to include creator notes.
As the approver (separate user) put the report in needs action. Include manager notes.
As the creator, revisit the report. Edit the creator notes. Resubmit.
As the approver: Confirm the notes have changed, approve the report. Once approved, confirm that the manager notes and additional notes have disappeared.

Confirm that all historical note data is gone.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-974
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1753


## Checklists

### Every PR


- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [n/a] UI review complete

### Production Deploy


### After merge/deploy
